test/test_dhcpv6.py

```diff
@@ -2,10 +2,42 @@
 # SPDX-License-Identifier: Apache-2.0
 
 from helpers import *
+from scapy.layers.dhcp6 import *
 
-def test_dhcpv6(prepare_ifaces):
-	DUID = DUID_LL(lladdr=VM1.mac)
+PXE_MODE = 0
+IPXE_MODE = 1
+
+class SubOptionField(Packet):
+	name = "SubOptionField"
+	fields_desc = [
+		ShortField("subopt_len", None),  # 2 bytes for suboption length
+		StrLenField("subopt_data", "", length_from=lambda pkt: pkt.subopt_len)
+	]
+	def extract_padding(self, p):
+		return "", p
+
+# Via library provided DHCP6OptUserClass doesnt consider suboption length, so extend it
+class DHCP6OptUserClassWithSubOption(DHCP6OptUserClass):
+	fields_desc = [
+		ShortEnumField("optcode", 15, dhcp6opts),
+		ShortField("optlen", None),
+		PacketListField("user_class_data", [], SubOptionField,
+						length_from=lambda pkt: pkt.optlen)
+	]
+	def post_build(self, p, pay):
+		if self.optlen is None:
+			length = len(p) - 4
+			p = p[:2] + struct.pack("!H", length) + p[4:]
+		return p + pay
```

Comment:
I don't even know what's going on here, but I'll trust you and the test results 😄

Reply:
Implementing the post_build from the parent class to calculate the length of this layer (if not done, already) and position it to third and fourth byte packed to network byte order.